make_helpers/build_macros.mk

```diff
@@ -312,7 +313,7 @@ bl${1}_dirs: | ${OBJ_DIRS}
 $(eval $(call MAKE_OBJS,$(BUILD_DIR),$(SOURCES),$(1)))
 $(eval $(call MAKE_LD,$(LINKERFILE),$(BL_LINKERFILE),$(1)))
 
-$(ELF): $(OBJS) $(LINKERFILE) | bl$(1)_dirs
+$(ELF): $(OBJS) $(LINKERFILE) | bl$(1)_dirs $(BL_LIBS)
```

Comment:
Hmm, this seems to be colliding with the PR#1394.

Reply:
It's not too invasive, I think. and that PR isn't ready anyway. This one is mostly ready.

Reply:
I was reviewing where these new BL*_LIBS macros are defined, but I couldn't find them. Is this patch preparing other patches?

Reply:
Actually we added this change for supporting library linkage on our own platform. It was added in the past when we started to implement Marvell platform on ATF-1.2. I was not aware of any other patches waiting for this change.

Reply:
If you don't need this now, it would be better to remove this to avoid conflict with the other work Soby/Roberto refer to.

Reply:
We need it for sure. Without this patch the early boot stage cannot be built. This stage is using DDR initialization library, which is out of TF-A tree. If I remove this patch, Marvell platform support in TF-A will be useless.

Reply:
OK, fair enough. We will rebase our other work on top of this.

Reply:
I forgot to mention that the actual usage of this library macros in in makefile of BLE:
https://github.com/MarvellEmbeddedProcessors/ble-marvell/blob/atf-mainline/ble.mk
We make additional  usage of library support in our LSP sources, but these changes are not yet mainlined.